From f6892df8eb115fa712fd69068432b48b028e1e54 Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd.kuhls@t-online.de>
Date: Sat, 20 Nov 2021 22:48:54 +0100
Subject: [PATCH] util/u_printf.h: fix build
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes build error with uClibc and gcc-9.3.0:

In file included from ../src/util/u_printf.cpp:24:
../src/util/u_printf.h:43:41: error: ‘va_list’ has not been declared
   43 | size_t u_printf_length(const char *fmt, va_list untouched_args);

Patch sent upstream:
https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/13897

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
---
 src/util/u_printf.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/util/u_printf.h b/src/util/u_printf.h
index 44dcce5529f..14f324de10c 100644
--- a/src/util/u_printf.h
+++ b/src/util/u_printf.h
@@ -25,6 +25,7 @@
 #ifdef __cplusplus
 
 #include <string>
+#include <stdarg.h>
 
 /* find next valid printf specifier in a C++ std::string */
 size_t util_printf_next_spec_pos(const std::string &s, size_t pos);
-- 
GitLab

